GVRf/Framework/jni/objects/components/transform.cpp

```diff
@@ -43,6 +42,16 @@ void Transform::invalidate() {
             (*it)->transform()->invalidate();
         }
     }
+    // scale rotation_ if needed to avoid overflow
```

Comment:
Looks like you do this test on every call to `invalidate()`, regardless of whether the `model_matrix_.isValid()` or not.

Seems like it would be better to do this in `Transform::getModelMatrix()`, if and only if the `Transform` has been invalidated.